src/common/log.c

```diff
@@ -156,24 +197,40 @@ static void log_callback(const zf_log_message* msg, void* param) {
 	}
 }
 
-void ardop_log_start(const bool enable_files) {
+void ardop_log_start(const bool enable_files, const bool syslog) {
 	zf_log_set_output_v(ZF_LOG_PUT_STD, NULL, log_callback);
-	ardop_log_enable_files(enable_files);
+	ardop_log_enable_files_and_console(enable_files, syslog);
 }
 
 void ardop_log_stop() {
-	ardop_log_enable_files(false);
+	ardop_log_enable_files_and_console(false, false);
 	zf_log_set_output_v(ZF_LOG_PUT_STD, NULL, log_callback_discard);
 }
 
 void ardop_log_enable_files(const bool file_output) {
+	ardop_log_enable_files_and_console(file_output, ArdopLogConsoleSyslog);
+}
+
+void ardop_log_enable_files_and_console(const bool file_output, const bool syslog) {
 	// Close all log files
 	// They are reopened on the next loggable message
 	for (size_t i = 0; i < sizeof(ALL_LOG_FILES) / sizeof(ALL_LOG_FILES[0]); ++i) {
 		ardop_logfile_close(ALL_LOG_FILES[i]);
 	}
 
 	ArdopLogFileEnabled = file_output;
+
+#ifdef LOG_OUTPUT_SYSLOG
+	if (ArdopLogConsoleSyslog) {
+		closelog();
+	}
+
+	ArdopLogConsoleSyslog = syslog;
+
+	if (ArdopLogConsoleSyslog) {
+		openlog(NULL, LOG_CONS | LOG_PERROR | LOG_PID, LOG_USER);
```

Comment:
`LOG_PERROR` also sends log messages to stderr, in addition to syslog. In systemd 255, this causes duplicate log lines **unless** the unit file contains

```ini
[Service]
StandardError=null
```

to ignore them.

It may be worth having a separate `--systemd` CLI argument which omits `LOG_PERROR`. We could also use this flag to configure other settings which we deem appropriate for daemonized / unattended operation. I don't imagine there will be too many of those. Maybe `--syslog --nologfile`?

Reply:
43939741f58024b5f06ffeb23412a83129302d4d writes to syslog only, not to stderr.